csclient/params/params.go

```diff
@@ -258,6 +262,30 @@ type WhoAmIResponse struct {
 	Groups []string
 }
 
+// Resource holds the metadata about a Resource defined for a charm.
+type Resource struct {
+	Name        string
+	Type        ResourceType
```

Comment:
Well... no, I mean, the type as a part of the API struct helps tell a consumer of the API what the API uses.  Yes, for our specific code, the code does all the right stuff... but I'm talking about the structs as a way to document the actual HTTP API.